gitmanager.py

```diff
@@ -341,12 +341,43 @@ def merge_pull_request(cls, pr_id, comment=""):
         if not response:
             raise ConnectionError("Cannot connect to GitHub API")
         pr_info = response.json()
-        if pr_info["user"]["login"] != "SmokeDetector":
-            raise ValueError("PR #{} is not created by me, so I can't approve it.".format(pr_id))
-        if "<!-- METASMOKE-BLACKLIST" not in pr_info["body"]:
-            raise ValueError("PR description is malformed. Blame a developer.")
         if pr_info["state"] != "open":
             raise ValueError("PR #{} is not currently open, so I won't merge it.".format(pr_id))
+        if pr_info["user"]["login"] != "SmokeDetector":
+            pr_sha = pr_info["head"]["sha"]
+            response = requests.get("https://api.github.com/repos/{}/pulls/{}/files"
+                                    .format(GlobalVars.bot_repo_slug, pr_id))
+            pr_files = response.json()
+            list_files = ["watched_keywords.txt", "watched_numbers.txt", "bad_keywords.txt",
+                          "blacklisted_numbers.txt", "blacklisted_usernames.txt", "blacklisted_websites.txt"]
+            line_changes = 0
+            for file in range(len(pr_files)):
+                line_changes += int(pr_files[file]["changes"])
+                if pr_files[file]["filename"] not in list_files:
+                    raise ValueError("PR #{} modifies files other than the list files, so I can't approve it."
+                                     .format(pr_id))
+            if line_changes > 6:
+                raise ValueError("PR #{} modifies more than 6 lines, so I can't approve it.".format(pr_id))
+            if not pr_info["mergeable"]:
+                raise ValueError("PR #{} is not mergeable, so I can't approve it.".format(pr_id))
+            github_checks = requests.get("https://api.github.com/repos/{}/pulls/{}/check-runs".
+                                         format(GlobalVars.bot_repo_slug, pr_id))
+            github_checks = github_checks.json()
+            github_check_count = github_checks["total_count"]
+            github_checks = github_checks["check_runs"]
+            for check in range(int(github_check_count)):
+                if github_checks[check]["conclusion"] != "success":
+                    raise ValueError("PR #{} has failed one or more GitHub Actions, so I can't approve it."
+                                     .format(pr_id))
+            ci_check = requests.get("https://api.github.com/repos/{}/commits/{}/status"
+                                    .format(GlobalVars.bot_repo_slug, pr_sha))
+            ci_check = ci_check.json()
+            ci_check = ci_check["statuses"]
+            for check in range(int(len(ci_check))):
+                if ci_check[check]["context"] == "ci/circleci: build":
```

Comment:
Only looking for a specific check feels potentially fragile, especially if checks are changed in the future (e.g., more added).  I think you're trying to exclude the PullApprove check here (which is correct, because the user approving it must be code-privileged, so they could approve anyway).  To do that, I'd personally just check if the status is _not_ the PullApprove check to exclude only that one, while still checking for potential future status providers.